requiam/commons.py

```diff
@@ -31,6 +34,59 @@ def figshare_stem(stem='', production=True):
     return stem_query
 
 
+def dict_load(config_file, vargs=None):
+    """
+    Purpose:
+      Read in a config INI file using configparser and return a dictionary
+      with sections and options
+
+    :param config_file: str. Full/relative path of configuration file
+    :param vargs: Dictionary containing command-line arguments
+    :return config_dict: dict of dict with hierarchy of sections follow by options
+    """
+
+    if vargs is None:
+        vargs = dict()
+
+    config = configparser.ConfigParser()
+    config.read(config_file)
+
+    # Read in default settings
+    config_dict = {}
+    for section in config.sections():
+        config_dict[section] = {}
+        for option in config.options(section):
+            option_input = config.get(section, option)
+            if option_input in ['True', 'False']:
+                config_dict[section][option] = config.getboolean(section, option)
+            else:
+                if option_input.isdigit():  # set as integer
+                    config_dict[section][option] = int(option_input)
+                else:
+                    config_dict[section][option] = config.get(section, option)
+
+    # Populate with command-line arguments overrides
+    config_dict['extras'] = {}
+    if vargs:
+        for p in vargs.keys():
+            if not isinstance(vargs[p], type(None)):
```

Comment:
Should this also be `vargs[p] is None` ?

Reply:
Can change but should be `if vargs[p] is not None:`